doc/rpc_marshalable_objects.md

```diff
@@ -7,12 +7,38 @@ StreamJsonRpc allows transmitting marshalable objects (i.e., objects implementin
 
 Marshalable interfaces must:
 
-1. Extend `IDisposable`.
+1. Extend `IDisposable` (unless interface is call-scoped).
 1. Not include any properties.
 1. Not include any events.
 
 The object that implements a marshalable interface may include properties and events as well as other additional members but only the methods defined by the marshalable interface will be available on the proxy, and the data will not be serialized.
 
+## Call-scoped vs. explicitly scoped
+
+### Explicit lifetime
+
+An RPC marshalable interface has an explicit lifetime by default.
+This means that the receiver of a marshaled object owns its lifetime, which may extend beyond an individual RPC call.
+Memory for the marshaled object and its proxy are not released until the receiver either disposes of its proxy or the JSON-RPC connection is closed.
+
+### Call-scoped lifetime
+
+A call-scoped interface produces a proxy that is valid only during the RPC call that delivered it.
+It may only be used as part of a method request as or within an argument.
+Using it as or within a return value or exception will result in an error.
+
+This is the preferred model when an interface is expected to only be used within request arguments because it mitigates the risk of a memory leak due to the receiver failing to dispose of the proxy.
+This model also allows the sender to retain control over the lifetime of the marshaled object.
+
+Opt into call-scoped lifetimes by setting the `CallScopedLifetime` property to `true` on the attribute applied to the interface:
+
+```css
+[RpcMarshalable(CallScopedLifetime = true)]
+```
+
+It is not possible to customize the lifetime of an RPC marshaled object except on its own interface.
```

Comment:
How does this work with inheritance? Can I override it on an inherited interface if I just use RpcMarshalable attribute on the inheriting interface.

Reply:
There is no inheritance of this attribute. Here is what I'm adding to the doc:

> The `RpcMarshalableAttribute` must be applied directly to the interface used as the return type, parameter type, or member type within a return type or parameter type's object graph.
The attribute is not inherited.
In fact different interfaces in a type hierarchy can have this attribute applied with distinct settings, and only the settings on the attribute applied directly to the interface used will apply.